spec/system/comments/user_fills_out_comment_spec.rb

```diff
@@ -46,4 +46,71 @@
     find(:xpath, "//div[contains(@class, 'reply-actions')]/input[@name='commit']").click
     expect(page).to have_text(raw_comment)
   end
+
+  # This is basically a black box test for
+  # ./app/javascripts/packs/validateFileInputs.js
+  # which is logic to validate file size and format when uploading via a form.
+  it "User attaches a valid image" do
+    visit article.path.to_s
+
+    wait_for_javascript
+
+    attach_file(
+      "image-upload-main",
+      Rails.root.join("app/assets/images/sloan.png"),
+      visible: false,
+    )
+
+    expect(page).to have_no_css("div.file-upload-error")
+  end
+
+  # This is basically a black box test for
+  # ./app/javascripts/packs/validateFileInputs.js
+  # which is logic to validate file size and format when uploading via a form.
+  it "User attaches a large image" do
+    visit article.path.to_s
+
+    wait_for_javascript
+
+    reduce_max_file_size = 'document.querySelector("#image-upload-main").setAttribute("data-max-file-size-mb", "0")'
+    page.execute_script(reduce_max_file_size)
+    expect(page).to have_selector('input[data-max-file-size-mb="0"]', visible: false)
```

Comment:
I'm not sure we need this assertion, isn't it asserting on test setup? Or was the purpose to fail if `reduce_max_file_size` fails so we don't get false positives?

Reply:
Correct, it's the latter. The purpose is to fail if `reduce_max_file_size` fails to prevent a false positive.